src/main/java/org/gooteam/Introduction.java

```diff
@@ -23,6 +26,12 @@ public void onMessage(GooMessage message) {
         String nickname = member.getNickname();
         Role role = event.getGuild().getPublicRole();
         TextChannel textChannel = event.getChannel().asTextChannel();
-        textChannel.sendMessage("hi " + nickname + ". im dr goo").queue();
+        hi-goo
+        if (nickname == null) {
+            textChannel.sendMessage("hi " + author.getName() + ". im dr goo").queue();
```

Comment:
wasnt aware of how queue worked, i guess its generally preferable?

https://github.com/discord-jda/JDA/wiki/7%29-Using-RestAction